app/controllers/users_controller.rb

```diff
@@ -23,13 +25,32 @@ def edit
   end
 
   def update
-    @user = current_user
-    ConfirmationSender.send_confirmation_to(@user)
-    redirect_to user_confirmation_path
+    @user = only_current_user
+    if @user.update_attributes(user_params)
+      @user.add_role if params[:user][:role_to_add]
+      flash[:success] = "Your account information has been updated!"
+      redirect_to user_path(@user)
+    else
+      render :edit
+    end
+    # ConfirmationSender.send_confirmation_to(@user)
```

Comment:
I think these two lines could be moved to a ConfirmationsController#create so we don't have to use an if/else here to deal with whether we are updating the password or something else.  What do you think @meganft ?